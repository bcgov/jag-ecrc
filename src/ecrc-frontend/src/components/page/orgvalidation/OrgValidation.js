import React, { useState } from "react";
import PropTypes from "prop-types";
import axios from "axios";
import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import OrgValidationText from "../../base/orgvalidationtext/OrgValidationText";
import SideCard from "../../base/sideCard/SideCard";
import "../page.css";

export default function OrgValidation({
  page: {
    setOrg,
    pageLayout: { header, sideCard1, sideCard2 }
  }
}) {
  const [orgInput, setOrgInput] = useState("");

  const orgValidation = () => {
    axios
      .get(`/ecrc/doAuthenticateUser?orgTicketId=${orgInput}`)
      .then(res => {
        setOrg(res.data.accessCodeResponse);
      })
      .catch();
  };

  const textInput = {
    label: "Access code",
    id: "orgId",
    textInputStyle: "placeHolder",
    isRequired: true
  };

  const button = {
    label: "Validate",
    buttonStyle: "btn btn-primary",
    buttonSize: "btn btn-sm",
    type: "submit"
  };

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content">
          <OrgValidationText
            textInput={textInput}
            onChange={setOrgInput}
            button={button}
            onClick={orgValidation}
          />
        </div>
        <div className="sidecard">
          <SideCard sideCard={sideCard1} />
          <SideCard sideCard={sideCard2} />
        </div>
      </div>
      <Footer />
    </main>
  );
}

OrgValidation.propTypes = {
  page: PropTypes.shape({
    setOrg: PropTypes.func.isRequired,
    pageLayout: PropTypes.shape({
      header: PropTypes.shape({
        name: PropTypes.string.isRequired
      }),
      sideCard1: PropTypes.shape({
        heading: PropTypes.string.isRequired,
        content: PropTypes.string.isRequired
      }),
      sideCard2: PropTypes.shape({
        heading: PropTypes.string.isRequired,
        content: PropTypes.string.isRequired
      })
    }).isRequired
  }).isRequired
};
