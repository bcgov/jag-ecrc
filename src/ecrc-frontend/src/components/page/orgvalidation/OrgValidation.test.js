import React from "react";
import { create } from "react-test-renderer";

import OrgValidation from "./OrgValidation";

describe("OrgValidation Component", () => {
  test("Matches the snapshot", () => {
    const header = {
      name: "Criminal Record Check"
    };

    const setOrg = () => {};

    const page = {
      setOrg,
      header
    };

    const orgValidationPage = create(<OrgValidation page={page} />);
    expect(orgValidationPage.toJSON()).toMatchSnapshot();
  });
});
