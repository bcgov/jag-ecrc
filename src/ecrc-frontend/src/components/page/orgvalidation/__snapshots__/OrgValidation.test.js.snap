// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgValidation Component Matches the snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="https://www2.gov.bc.ca"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div
          className="navbar-brand"
        >
          Criminal Record Check
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav"
          >
            <a
              className="nav-item nav-link"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <div>
        <p>
          If you work or plan to work at an organization that works with children and/or vulnerable adults in BC, you need to apply and consent to a criminal record check. This portal enables you to easily apply to a criminal record check under the
           
          <a
            href="http://www.bclaws.ca/EPLibraries/bclaws_new/document/ID/freeside/00_96086_01"
          >
            Criminal Records Review Act (CRRA).
          </a>
        </p>
        <div>
          <div>
            <h3>
              I'm ready
            </h3>
            <p>
              To submit an online request for a criminal record check, you must:
            </p>
            <ul>
              <li>
                Be at least 12 years of age as of today's date
              </li>
              <li>
                Have an access code provided by your organization
              </li>
              <li>
                Have your identity verified through an Electronic Verification Identity (EIV) process.
                <p
                  style={
                    Object {
                      "fontStyle": "italic",
                    }
                  }
                >
                  <strong>
                     NOTE. 
                  </strong>
                  Not all individuals will be able to use an EIV process and may be required to submit the request to a criminal record check through a manual process.
                </p>
              </li>
            </ul>
            <div
              style={
                Object {
                  "float": "left",
                  "maxWidth": "350px",
                }
              }
            >
              <p>
                Enter the access code provided by your organization. An access code is required to proceed with the online submission:
              </p>
            </div>
            <div
              style={
                Object {
                  "float": "left",
                }
              }
            >
              <div>
                <div>
                  <div>
                    <label
                      className="textinput_label"
                      htmlFor="orgId"
                    >
                      Access code
                    </label>
                    <span
                      className="mandatory"
                      id="asterisk"
                    >
                      *
                    </span>
                  </div>
                  <input
                    className="textinput_editable_white"
                    id="orgId"
                    onChange={[Function]}
                    type="text"
                  />
                </div>
              </div>
              <div>
                <strong>
                  Access code must be entered to proceed *
                </strong>
              </div>
              <p />
              <div
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <button
                  className="btn btn-primary btn btn-sm"
                  onClick={[Function]}
                  type="submit"
                >
                  Validate
                </button>
              </div>
            </div>
            <div
              style={
                Object {
                  "clear": "both",
                }
              }
            />
          </div>
          <br />
          <div>
            <h3>
              I need more information
            </h3>
            <ul>
              <li>
                <a
                  href="volunteer"
                >
                  I'm an employee or a volunteer
                </a>
                 and I want to know
                 
                <a
                  href="http://www.rcmp-grc.gc.ca/en/types-criminal-background-checks"
                >
                  why I need to apply for a criminal record check
                </a>
              </li>
              <li>
                <a
                  href="https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check/organization-registration/employee-organization-registration/employee-contact-registration"
                >
                  I'm an authorized contact
                </a>
                 
                who is responsible for facilitating the criminal record check for my organization
              </li>
              <li>
                <a
                  href="https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check/employer-organizations"
                >
                  I'm an employer organization
                </a>
                 
                and I want to learn more about registering with the Criminal Record Review Program (CRRP)
              </li>
              <li>
                <a
                  href="https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check/volunteer-organizations"
                >
                  I'm a volunteer organization
                </a>
                 
                and I want to register with the CRRP
              </li>
            </ul>
          </div>
          <br />
          <div>
            <h3>
              I need help
            </h3>
            <p>
              To get your organization access code, contact
               
              <strong
                style={
                  Object {
                    "fontStyle": "italic",
                  }
                }
              >
                your organization
              </strong>
            </p>
            <p>
              For questions about the online submission process or to register your organization, contact the Criminal Record Review Program (CRRP) at
               
              <a
                href="mailto:criminalrecords@gov.bc.ca"
              >
                criminalrecords@gov.bc.ca
              </a>
               
              or by phone at 1-855-587-0185 (Option 2).
            </p>
          </div>
          <br />
        </div>
      </div>
    </div>
    <div
      className="sidecard"
    >
      <div>
        <div
          className="dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="bluegrey-container"
                id="bluegrey-section"
              >
                <div
                  className="container-background bluegrey-heading"
                >
                  <h2
                    className="heading-style"
                  >
                    Need help with an access code?
                  </h2>
                </div>
                <div
                  className="bluegrey-content"
                >
                  <p
                    className="content-style"
                  >
                    Access code is specific to each organization. If you need help with the organization access code, please 
                    <b
                      className="block"
                    >
                      contact your organization.
                    </b>
                  </p>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  Why do I need to apply and consent to a criminal record check?
                </h2>
                <div
                  className="submit-content"
                >
                  <p>
                    Learn more about the process
                    <button
                      className="btn btn-primary dark-button btn btn-md"
                      onClick={[Function]}
                      type="submit"
                    >
                      VISIT THE CRIMINAL RECORD REVIEW WEBSITE
                    </button>
                  </p>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
